### DISCUSSION TITLE
Persist state on web3

### DISCUSSION THREAD

**Discussion by Tails** [DustinBrett/daedalOS#50] [Feature Requests]
This is a really cool project and it would be nice to be able to "login" to such a webtop from anywhere. 

I imagine that on exit the IndexedDb contents could be synced to Sia SkyNet, StorJ, Filecoin or others (encrypted) and that one can retrieve their private state using a login screen.

**Comment by DustinBrett** [comment]
Thanks for the suggestion! I will add it to my list as Web3 is the rage now. I know https://js.ipfs.io/ was something I had been considering for this. I want to keep as much as possible on the client side.

**Comment by Tails** [comment]
IPFS is a possibility, but the downside of it would be that there is no guarantee your data would stay hosted unless the user would also host their own server or pay for Pinata (pinning service), which kind of defeats the purpose. That's why I mentioned SkyNet since there is a minimum uptime and free tier guarantee.

**Comment by DustinBrett** [comment]
Ah cool, thanks for the tip. I am new to Web3, but will look into it more.

**Comment by DustinBrett** [comment]
Surprised at 3 down votes to the idea. I wonder is the issue Web3? Do people want persistent storage? What would it be used for? Why would it be a good/bad idea?

**Comment by markus-li** [comment]
Persistent storage would be great, but maybe just as a sync to a node.js instance running on a self-hosted server? The backend for storing could be either filesystem directly, or maybe even a database for parts which are not very large.
What would also be nice within the space of persistence would be to let files sync bidirectionally with a folder on a server running a small node.js-based service. All to make it easy to self-host. Been looking at your very nicely created web desktop as an option for a project of mine, if the need arises I might add the syncing to a self-hosted node.js instance at some point if I can just find the time.

**Reply by DustinBrett** [reply]
Cool idea! Thanks for the input. That does sound like a good option. I'd like to be able to add a BrowserFS backend as that would integrate nicely with what I am using already. I've added this to my to do board.

https://github.com/jvilk/BrowserFS#backends

**Reply by markus-li** [reply]
That looks like a fairly good way of doing it since it is already very much extensible. Very neat!
